tests/resources/mlflow-test-plugin/setup.py

```diff
@@ -8,7 +8,7 @@
     packages=find_packages(),
     # Require MLflow as a dependency of the plugin, so that plugin users can simply install
     # the plugin & then immediately use it with MLflow
-    install_requires=["mlflow"],
+    install_requires=[],  # Remove mlflow dependency, it can cause tests to run against pypi
```

Comment:
@harupy @smurching Didn't this come up in the context of another ticket? Can you comment on how to best resolve it?

Reply:
It did yeah - we were confused why e.g. `gorilla` was still being installed in our test environment despite the fact that we'd removed it as a dependency from `setup.py`, e.g. in https://github.com/mlflow/mlflow/pull/3666/checks?check_run_id=1382567094#step:3:308. It turns out in our dependency installation scripts for Github actions, we install this test plugin and then install MLflow from source - because we install the test plugin first, we end up installing `mlflow` from pypi as a transitive dependency of the plugin, and then overwriting that installation with MLflow from source.

The change here avoids that problem by removing `mlflow` as a dependency from the test plugin. This seems ok as we ought to only use the test plugin in our CI - in practice though, we reference the test plugin in our [docs on writing plugins](https://mlflow.org/docs/latest/plugins.html#defining-a-plugin). 

To get around this we could either:
a) Define another example plugin to reference in our docs & add tests verifying that it works (seems like more work)
b) See if we can do something like `install_requires=[os.path.join(".", os.pardir, os.pardir, os.pardir)]` to depend on MLflow installed from source (reduces the value of the test plugin as an example)
c) Just update our install-common-deps.sh CI script to [install MLflow from source](https://github.com/mlflow/mlflow/blob/eda5f435c65d8e7ceab4d4ba82e7e18b2e2cdef3/dev/install-common-deps.sh#L51) before we [install other Python dependencies](https://github.com/mlflow/mlflow/blob/eda5f435c65d8e7ceab4d4ba82e7e18b2e2cdef3/dev/install-common-deps.sh#L33-L49), including the test plugin

Reply:
IMO option a) is maybe the best actual approach, but I'd be fine with option c) too; the risk is that one of our test requirements has a strict version dependency on `mlflow` and ends up overwriting our source installation of `mlflow`, but this seems unlikely / we could potentially test for this

Reply:
As discussed offline with @smurching , I think the concern here is that depending on `mlflow` (even a source-installed copy) will bring in dependencies that aren't included by the skinny client, degrading the utility of our skinny client tests.

As a result, I think it's best for us to adopt option (a) here. @eedeleon Is this something you can do as part of this PR? If not, I can pitch in.

Reply:
I think we could implement a like b.

requirements = [] if MLFLOW_TEST_ENV in os.environ else ["mlflow"]

Reply:
We can add a test that verifies MLFLOW_TEST_ENV(we can use any env var) is set as well

Reply:
That `os.environ` code will show up in our example reference too, though, which we probably don't want.

Reply:
> We can add a test that verifies MLFLOW_TEST_ENV(we can use any env var) is set as well

Agreed!